### PR TITLE
protobuf: add XPub message and conversion function

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: messages/common.options [hunk 1]
```diff
@@ -14,3 +14,7 @@
 
 PubResponse.pub max_size:500
 RootFingerprintResponse.fingerprint fixed_length:true max_size:4
+XPub.depth fixed_length:true max_size:1
+XPub.parent_fingerprint fixed_length:true max_size:4
+XPub.chain_code fixed_length:true max_size:32
+XPub.public_key fixed_length:true max_size:33
```
Comment (NickeZ) [inline messages/common.options]: Why 33 (please write a comment)? and missing newline.

File: messages/common.proto [hunk 2]
```diff
@@ -24,3 +24,11 @@ message RootFingerprintRequest {
...
   bytes fingerprint = 1;
 }
+
+message XPub {
+  bytes depth = 1;
+  bytes parent_fingerprint = 2;
+  uint32 child_num = 3;
+  bytes chain_code = 4;
+  bytes public_key = 5;
+}
```
Comment (NickeZ) [inline messages/common.proto]: missing newline.

File: src/apps/btc/btc_common.c [hunk 3]
```diff
@@ -359,3 +359,33 @@ bool btc_common_pkscript_from_outputhash(
...
 }
+
+bool btc_common_xpub_from_protobuf(const XPub* xpub_in, struct ext_key* xpub_out)
+{
+    struct ext_key* tmp = NULL;
+    // libwally quirk: bip32_key_init_alloc takes 20 bytes for the parent fingerprint but only uses
+    // the first 4 bytes (the real size of the fingerprint).
+    uint8_t parent_fingerprint[20] = {0};
+    memcpy(parent_fingerprint, xpub_in->parent_fingerprint, sizeof(xpub_in->parent_fingerprint));
+    // Another libwally quirk: no function to create an ext_key from params without malloc.
```
Comment (NickeZ) [inline src/apps/btc/btc_common.c]: did you report these issues to libwally repo?

File: src/apps/btc/btc_common.h [hunk 4]
```diff
@@ -132,4 +132,9 @@ USE_RESULT bool btc_common_pkscript_from_outputhash(
     uint8_t* pk_script,
     size_t* pk_script_len);
 
+/**
+ * Converts between an xpub provided via a protobuf message to a libwally xpub.
+ */
+USE_RESULT bool btc_common_xpub_from_protobuf(const XPub* xpub_in, struct ext_key* xpub_out);
```
Comment (NickeZ) [inline src/apps/btc/btc_common.h]: Should this really live in the "btc" namespace? Doesn't seem only related to BTC.

File: messages/common.options (hunk 1 above)
Comment (benma) [reply messages/common.options]: Done

File: messages/common.proto (hunk 2 above)
Comment (benma) [reply messages/common.proto]: Done

File: src/apps/btc/btc_common.c (hunk 3 above)
Comment (benma) [reply src/apps/btc/btc_common.c]: Just did. I have a whole bunch of issues, I think we need to provide the PRs ourselves though.

**Pull request merged**